### PR TITLE
Merging initial drive base to main

### PR DESCRIPTION
**Pull request by NogaRoze**
You were all there.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java:80]: Consider making this a string value, i.e. Drive Mode: "Tank" or "Arcade". More descriptive, and extendable (in case we want to add Swerve :P )

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: Would be useful to comment why these motors are inverted

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: Class description needs to be updated

File: src/main/java/frc/robot/RobotContainer.java
Comment (azvanderpas) [inline src/main/java/frc/robot/RobotContainer.java]: For the moment we should probably replace this with 'null' or an empty command object.

File: README.md
Comment (azvanderpas) [inline README.md]: Might be narcissistic, but I'd like my name in the README :)

**Review by azvanderpas — CHANGES_REQUESTED** [review]
In addition to my comments, ideally we would test the code on a robot before finishing the merge. Maybe we can get @AndrewIverson to test it when he's at the shop on Thursday?

**Comment by NogaRoze** [comment]
Addressed all current comments.

File: src/main/java/frc/robot/Constants.java
Comment (AndrewIverson) [inline src/main/java/frc/robot/Constants.java]: Lets remove the _ from the constant names, use camel case.

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: This sort of copy and paste code is where making an array of CANSparkMax objects and then using loops would be helpful.

File: src/main/java/frc/robot/Constants.java
Comment (azvanderpas) [inline src/main/java/frc/robot/Constants.java]: Consider kButtonA instead so you don't have adjacent capital letters

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java:19]: This comment should add more detail to what DriveBaseSubsystem does/is.
Ideally, it would be a description of the drive base it is meant to control and what devices, sensors, etc it uses. Other features like drive mode support and brief descriptions of the joystick mappings for each mode. 
Imagine next year coming back to the code and trying to remember what you did, and write down here what you think you might forget about how it is meant to work.

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: Robot.isReal() might be better here just for readability.

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: As above it could be made simpler if the CANEncoders were an array that could be iterated over.

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [reply src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: Created a new issue #9 to track, because I don't think this should block the PR and should be a future upgrade instead.

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [reply src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: Created a new issue #9 to track, because I don't think this should block the PR and should be a future upgrade instead.

File: src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java]: ExternalFollower isn't used anymore, this import can be removed

**Review by azvanderpas — APPROVED** [review]

**Pull request merged**
